autogen/agentchat/groupchat.py

```diff
@@ -174,9 +175,29 @@ def append(self, message: Dict, speaker: Agent):
         message["content"] = content_str(message["content"])
         self.messages.append(message)
 
-    def agent_by_name(self, name: str) -> Agent:
-        """Returns the agent with a given name."""
-        return self.agents[self.agent_names.index(name)]
+    def agent_by_name(
```

Comment:
You should add a test for this function and make sure you test all edge cases with 100% coverage. It is too easy to make a simple mistake that will be hard to catch later.

Reply:
Tests added!

Reply:
what is the test coverage? are there any lines missed?